### ISSUE TITLE
0x809305a7 gmText needs some fixing

### ISSUE THREAD

**Issue by NoseyNick** [artemis-nerds/protocol-docs#129]
The docs call this one GameMasterMessagePacket. Officially it's gmText

Doc says "New as of v2.4.0" but v2.0.0 had a perfectly good GM console that sent gmText packets just fine, and in fact v2.4.0 introduced more fields that are NOT in the doc. These packets may even have existed before that, but I have no v1.X to confirm (donations of zipfiles welcome, for archaeological purposes!). The fields have changed a bit over time, and have recently been (ab)used for the freeform ship-to-ship comms text to create idleText packets ( see #123 )

Packet Type 0x809305a7 = JamCRC("gmText") - from client

Payload

Recipients list (new in v2.4.0, see format below)
- Before v2.4.0, this did not exist.
- In v2.4.0 to v2.5.x, this was a set of 8 1-byte booleans, one per player ship (Artemis, Intrepid, Aegis, etc) set to 1 for any ship that should receive this gmText and 0 for those that should not.
- In v2.6.0 this became a list of 0 to N ship ObjectIDs (integers), terminated by an integer 0.

Console type (int, enumeration), as described in the doc (except I don't think it was ever a byte?), and note new console type 3000 = freeform ship-to-ship comms introduced in v2.7.0 ( which the server will send back as "idleText" - see #123 )

Sender (string), as described in the doc

Message (string), as described in the doc

Priority (new in v2.7.0, OPTIONAL int) approx as described in #128 except 0="General", 1="Distress", 2="Side MIssion" (close enough), 4="Station Status" (presumably supposed to be the same). Seems to NOT be included for "Console type 3000 ship-to-ship freeform text" but IS included for real gmTexts from a real GM?

**Comment by rjwut** [comment]
Is the priority field included for popups? Seems like it shouldn't, but your description only excludes the 3000 console.

**Comment by NoseyNick** [comment]
Seems to be included for all the popups in 2.7.0, yup, probably everything from the actual GM, but NOT from the comms Text field?
```pl
gmText(1002,0, 1, UTF16("from") => UTF16("message side"), 2); # v2.7 Main SideMission
gmText(1002,0, 1, UTF16("from") => UTF16("message player"), 8); # v2.7 Main Player
gmText(1002,0, 1, UTF16("from") => UTF16("message distress"), 1); # v2.7 Main Distress
gmText(1002,0, 1, UTF16("from") => UTF16("message general to main"), 1); # v2.7 Main Distress
gmText(1002,0, 2, UTF16("from") => UTF16("message player"), 8); # v2.7 Helm Player
gmText(1002,0, 2, UTF16("from") => UTF16("message general to helm"), 1); # v2.7 Helm Distress
gmText(1002,0, 3, UTF16("from") => UTF16("message player"), 8); # v2.7 Weap Player
gmText(1002,0, 3, UTF16("from") => UTF16("message general to weap"), 1); # v2.7 Weap Distress
gmText(1002,0, 4, UTF16("from") => UTF16("message player"), 8); # v2.7 Eng Player
gmText(1002,0, 4, UTF16("from") => UTF16("message general to eng"), 1); # v2.7 Eng Distress
gmText(1002,0, 5, UTF16("from") => UTF16("message player"), 8); # v2.7 Sci Player
gmText(1002,0, 5, UTF16("from") => UTF16("message general to sci"), 1); # v2.7 Sci Distress
gmText(0, 3000, UTF16("Artemis") => UTF16("comms message from mainscreen")); # v2.6 idleTextFreeform
gmText(1003,0, 3000, UTF16("Artemis") => UTF16("Hello from the comms screen")); # v2.6 idleTextFreeform
```
... or in hex (DEADBEEF stripped):
```
a7059380ea030000000000000100000005000000660072006f006d0000000d0000006d0065007300730061006700650020007300690064006500000002000000 # 2.7.0 payload
a7059380ea030000000000000100000005000000660072006f006d0000000f0000006d00650073007300610067006500200070006c006100790065007200000008000000 # 2.7.0 payload
a7059380ea030000000000000100000005000000660072006f006d000000110000006d00650073007300610067006500200064006900730074007200650073007300000001000000 # 2.7.0 payload
a7059380ea030000000000000100000005000000660072006f006d000000180000006d006500730073006100670065002000670065006e006500720061006c00200074006f0020006d00610069006e00000001000000 # 2.7.0 payload
a7059380ea030000000000000200000005000000660072006f006d0000000f0000006d00650073007300610067006500200070006c006100790065007200000008000000 # 2.7.0 payload
a7059380ea030000000000000200000005000000660072006f006d000000180000006d006500730073006100670065002000670065006e006500720061006c00200074006f002000680065006c006d00000001000000 # 2.7.0 payload
a7059380ea030000000000000300000005000000660072006f006d0000000f0000006d00650073007300610067006500200070006c006100790065007200000008000000 # 2.7.0 payload
a7059380ea030000000000000300000005000000660072006f006d000000180000006d006500730073006100670065002000670065006e006500720061006c00200074006f0020007700650061007000000001000000 # 2.7.0 payload
a7059380ea030000000000000400000005000000660072006f006d0000000f0000006d00650073007300610067006500200070006c006100790065007200000008000000 # 2.7.0 payload
a7059380ea030000000000000400000005000000660072006f006d000000170000006d006500730073006100670065002000670065006e006500720061006c00200074006f00200065006e006700000001000000 # 2.7.0 payload
a7059380ea030000000000000500000005000000660072006f006d0000000f0000006d00650073007300610067006500200070006c006100790065007200000008000000 # 2.7.0 payload
a7059380ea030000000000000500000005000000660072006f006d000000170000006d006500730073006100670065002000670065006e006500720061006c00200074006f002000730063006900000001000000 # 2.7.0 payload
a705938000000000b80b00000800000041007200740065006d006900730000001e00000063006f006d006d00730020006d006500730073006100670065002000660072006f006d0020006d00610069006e00730063007200650065006e000000 # 2.7.0 payload
a7059380eb03000000000000b80b00000800000041007200740065006d006900730000001c000000480065006c006c006f002000660072006f006d002000740068006500200063006f006d006d0073002000730063007200650065006e000000 # 2.7.0 payload
```

**Comment by NoseyNick** [comment]
Another note... as a bitmask (values 1,2,4,8,16, etc) it feels like you ought to be able to select multiple priorities (EG 3 = 1 "Distress" +  2 "Side MIssion"), however the GM screen will only let you select one priority, corresponding to 1 bit OR 0="General".
